lib/ClangImporter/ClangImporter.cpp

```diff
@@ -631,6 +631,17 @@ importer::getNormalInvocationArguments(
       });
     }
 
+    // To support -apple-none triples, which are non-Darwin, we need to #define
+    // a few things that the Apple SDKs expect.
+    if (triple.getVendor() == llvm::Triple::VendorType::Apple) {
+      invocationArgStrs.insert(invocationArgStrs.end(),
+                               {"-D__APPLE__", "-D__MACH__"});
```

Comment:
mach should only be defined when the triple contains mach:
technically someone could write `xxx-apple-none-elf` (granted that would be rather strange...)

Reply:
+1 this should be conditional object file format == MachO

Reply:
Fixed.